server/src/lib/basket.ts

```diff
@@ -11,80 +11,154 @@ export const API_URL =
+    WHERE client_id = ?`,
+    [client_id]
+  );
+
+  return row;
+}
+
+function updateLastActive(currentUserStats: any) {
+  db.query(
+    `UPDATE user_client_newsletter_prefs
+      SET last_active = ?
+    WHERE client_id = ?`,
+    [currentUserStats.new_last_active, currentUserStats.client_id]
+  );
+
+  const data = {
+    email: currentUserStats.email,
+    last_active_date: toISO(currentUserStats.new_last_active),
+  };
+
+  sendRequest({
+    ...basketConfig,
+    form: data,
+  }).catch(err => console.error(err.message));
+
+  console.log('basket', JSON.stringify(data, null, 2));
+  return;
+}
+
+async function updateFullBasket(currentUserStats: any) {
+  const [[computed]] = await db.query(
     `
       SELECT
         email,
         basket_token,
         LEAST(
-          (SELECT MIN(clips.created_at) FROM clips WHERE client_id = user_clients.client_id),
-          (SELECT MIN(votes.created_at) FROM votes WHERE client_id = user_clients.client_id)
+          IFNULL((SELECT MIN(clips.created_at) FROM clips WHERE client_id = newsletter.client_id), NOW()),
+          IFNULL((SELECT MIN(votes.created_at) FROM votes WHERE client_id = newsletter.client_id), NOW())
         ) AS first_contribution_date,
         current_goal.created_at AS goal_created_at,
         current_goal.days_interval,
         MAX(awards.created_at) AS goal_reached_at,
-        NOW() AS last_active_date,
         (
           NOT EXISTS(
             SELECT 1
             FROM reached_goals
-            WHERE type = 'streak' AND client_id = user_clients.client_id
+            WHERE type = 'streak' AND client_id = newsletter.client_id
               AND count = 3  
           ) AND EXISTS(
             SELECT 1
             FROM streaks
-            WHERE client_id = user_clients.client_id
+            WHERE client_id = newsletter.client_id
               AND DATEDIFF(NOW(), started_at) >= 2
               AND DATEDIFF(NOW(), last_activity_at) <= 1
           )
         ) AS two_day_streak
-      FROM user_clients
-      LEFT JOIN custom_goals goals ON user_clients.client_id = goals.client_id
+      FROM user_client_newsletter_prefs AS newsletter
+      LEFT JOIN custom_goals goals ON newsletter.client_id = goals.client_id
                                       AND goals.locale_id = 1
       LEFT JOIN custom_goals current_goal ON (
-        user_clients.client_id = current_goal.client_id AND
+        newsletter.client_id = current_goal.client_id AND
         current_goal.created_at >= goals.created_at
       )
       LEFT JOIN awards ON current_goal.id = awards.custom_goal_id
-      WHERE user_clients.client_id = ? AND has_login
-      GROUP BY user_clients.client_id
+      WHERE newsletter.client_id = ?
+      GROUP BY newsletter.client_id
     `,
-    [client_id]
+    [currentUserStats.client_id]
   );
-  if (
-    !row ||
-    !row.basket_token ||
-    (!row.first_contribution_date && !row.goal_created_at)
-  ) {
-    return;
-  }
```

Comment:
Took out this whole block because: 
* If the user doesn't exist in `user_client_newsletter_prefs` they don't get emails and this whole sync function ends, which is already handled on line 148
* Sync function gets triggered for new goals, recording a clip, and submitting a vote. By definition, `first_contribution_date` and `goal_created_at` will never be empty at the same time, because the act of creating one of those is what leads to this function